### ISSUE TITLE
Fix `getaddrinfo ENOTFOUND challenge-keycloak`

### ISSUE THREAD

**Issue by tschaffter** [Sage-Bionetworks/challenge-registry#308]
Making this get request generates an error. Keycloak should not be involved in this request so why is this message shown?

```console
@serviceRegistryHost = http://localhost:8081
@apiGatewayHost = http://localhost:8082
@userServiceHost = http://localhost:8083
@keycloakHost = http://localhost:8080

### Check user service info via the API gateway

GET {{apiGatewayHost}}/user/actuator/info
```

![image](https://user-images.githubusercontent.com/3056480/173157041-d1bbbab2-8177-4a62-968d-d740263f07a3.png)

Making the request directly to the user service works:

```console
@serviceRegistryHost = http://localhost:8081
@apiGatewayHost = http://localhost:8082
@userServiceHost = http://localhost:8083
@keycloakHost = http://localhost:8080

### Check user service info

GET {{userServiceHost}}/actuator/info

HTTP/1.1 200 
Content-Type: application/vnd.spring-boot.actuator.v3+json
Transfer-Encoding: chunked
Date: Fri, 10 Jun 2022 22:07:09 GMT
Connection: close

{
  "application": {
    "name": "challenge-user-service"
  }
}
```

**Comment by tschaffter** [comment]
Actually KC is indeed used by the API gateway, see [Enable Oauth2.0 With Keycloak For Spring Cloud API Gateway](https://javatodev.com/microservices-authentication-and-authorization-with-keycloak/)

**Comment by tschaffter** [comment]
Error thrown by the AG when querying it:

```console
2022-06-10 22:04:39.375 ERROR 1 --- [or-http-epoll-6] a.w.r.e.AbstractErrorWebExceptionHandler : [74b0a0dc-14]  500 Server Error for HTTP GET "/user/actuator/info"

io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:8083
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
        *__checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.authorization.AuthorizationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.authorization.ExceptionTranslationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.authentication.logout.LogoutWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.savedrequest.ServerRequestCacheWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.context.SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.ui.LogoutPageGeneratingWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.ui.LoginPageGeneratingWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.authentication.AuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.oauth2.client.web.server.authentication.OAuth2LoginAuthenticationWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.oauth2.client.web.server.OAuth2AuthorizationRequestRedirectWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.context.ReactorContextWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.header.HttpHeaderWriterWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.config.web.server.ServerHttpSecurity$ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
        *__checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
        *__checkpoint ⇢ HTTP GET "/user/actuator/info" [ExceptionHandlingWebHandler]
Original Stack Trace:
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
        at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995) ~[netty-common-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.77.Final.jar!/:4.1.77.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.77.Final.jar!/:4.1.77.Final]
        at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]
```

**Comment by tschaffter** [comment]
## TODO

- [ ] Pull the API gateway out of the docker network and try the queries again.
- [ ] Try to access the logs of the extension `REST Client` to get more information.

**Comment by tschaffter** [comment]
Solved via #309
